internal/controller/management_controller.go

```diff
@@ -100,6 +101,59 @@ func (r *ManagementReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	return r.update(ctx, management)
 }
 
+func (r *ManagementReconciler) getRequestedProvidersList(ctx context.Context, management *kcm.Management) ([]kcm.Provider, error) {
+	list := slices.Clone(management.Spec.Providers)
+
+	existingProviders := make(map[string]struct{}, len(list))
+	for _, provider := range list {
+		existingProviders[provider.Name] = struct{}{}
+	}
+
+	var objList kcm.PluggableProviderList
+
+	if err := r.Client.List(ctx, &objList, &client.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labels.Set{
+			kcm.GenericComponentNameLabel: kcm.GenericComponentLabelValueKCM,
+		}),
+	}); err != nil {
+		return nil, err
+	}
+
+	for _, el := range objList.Items {
+		if _, exists := existingProviders[el.Name]; !exists {
+			list = append(list,
+				kcm.Provider{
+					Name:      el.Name,
+					Component: el.Spec.Component,
+				},
+			)
+
+			existingProviders[el.Name] = struct{}{}
+		}
+	}
+
+	slices.SortFunc(list, func(a, b kcm.Provider) int {
+		return strings.Compare(strings.ToLower(a.Name), strings.ToLower(b.Name))
+	})
+
+	return list, nil
+}
+
+func (r *ManagementReconciler) ensureRequestedProvidersList(ctx context.Context, management *kcm.Management) error {
+	rprov, err := r.getRequestedProvidersList(ctx, management)
+	if err != nil {
+		return err
+	}
+
+	if equality.Semantic.DeepEqual(rprov, management.Status.RequestedProviders) {
+		return nil
+	}
+
+	management.Status.RequestedProviders = rprov
+
+	return r.updateStatus(ctx, management)
```

Comment:
potential conflict: there will be 2 updates to the same resource version within a single event

I suggest removing it

Reply:
Correct, it updates the status and ends the whole reconciliation until new changes. The proposed solution has the different behavior

Reply:
Resolved [here](https://github.com/k0rdent/kcm/pull/1434/files#diff-7fbd16e56d0b84de5285626a796c9df0598bebbc52d3ef34bcbc9cf84e8e8c91R150)